abci/example/kvstore/kvstore.go

```diff
@@ -166,34 +167,64 @@ func (app *Application) InitChain(_ context.Context, req *types.InitChainRequest
 // - `=` is not the first or last byte.
 // - if key is `val` that the validator update transaction is also valid.
 func (app *Application) CheckTx(_ context.Context, req *types.CheckTxRequest) (*types.CheckTxResponse, error) {
+	code := CodeTypeOK
 	// If it is a validator update transaction, check that it is correctly formatted
 	if isValidatorTx(req.Tx) {
 		if _, _, _, err := parseValidatorTx(req.Tx); err != nil {
-			if app.useLanes {
-				return &types.CheckTxResponse{Code: CodeTypeInvalidTxFormat, Lane: app.lanes["val"]}, nil
-			}
-			return &types.CheckTxResponse{Code: CodeTypeInvalidTxFormat}, nil
+			code = CodeTypeInvalidTxFormat
 		}
 	} else if !isValidTx(req.Tx) {
-		return &types.CheckTxResponse{Code: CodeTypeInvalidTxFormat}, nil
+		code = CodeTypeInvalidTxFormat
 	}
 
 	if !app.useLanes {
-		return &types.CheckTxResponse{Code: CodeTypeOK, GasWanted: 1}, nil
+		return &types.CheckTxResponse{Code: code, GasWanted: 1}, nil
+	}
+
+	lane := uint32(0)
+	if code == CodeTypeOK {
+		lane = app.assignLane(req.Tx)
 	}
-	// Assign a lane to the transaction deterministically.
-	var lane uint32
-	txHash := tmhash.Sum(req.Tx)
+
+	return &types.CheckTxResponse{Code: code, GasWanted: 1, Lane: lane}, nil
+}
+
+// assignLane deterministically computes a lane for the given tx.
+func (app *Application) assignLane(tx []byte) uint32 {
+	if isValidatorTx(tx) {
+		return app.lanes["val"] // priority 9
```

Comment:
Maybe leave it above as a form of early return in this case.

Reply:
Refactored CheckTx, so I think this is not needed.